catroid/src/androidTest/java/org/catrobat/catroid/test/catblocks/AdvancedModeSettingsTest.kt

```diff
@@ -0,0 +1,105 @@
+import org.catrobat.catroid.content.Script
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.content.StartScript
+import org.catrobat.catroid.content.bricks.ChangeXByNBrick
+import org.catrobat.catroid.content.bricks.IfLogicBeginBrick
+import org.catrobat.catroid.content.bricks.SetXBrick
+import org.catrobat.catroid.ui.SpriteActivity
+import org.catrobat.catroid.ui.recyclerview.fragment.CatblocksScriptFragment
+import org.catrobat.catroid.ui.settingsfragments.SettingsFragment
+import org.catrobat.catroid.uiespresso.util.UiTestUtils
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class AdvancedModeSettingsTest {
+
+    @get:Rule
+    var baseActivityTestRule = FragmentActivityTestRule(
+        SpriteActivity::class.java,
+        SpriteActivity.EXTRA_FRAGMENT_POSITION,
+        SpriteActivity.FRAGMENT_SCRIPTS
+    )
+
+    @Before
+    fun setUp() {
+        CatblocksScriptFragment.testingMode = true
+        SettingsFragment.setCatBlocksAdvancedMode(ApplicationProvider.getApplicationContext(), true)
+        createProject()
+        baseActivityTestRule.launchActivity()
+    }
+
+    @After
+    fun tearDown() {
+        SettingsFragment.setCatBlocksAdvancedMode(ApplicationProvider.getApplicationContext(), false)
+        baseActivityTestRule.finishActivity()
+    }
+
+    @Test
+    fun testAdvancedModeSetting() {
+        val webViewUtils = WebViewUtils(baseActivityTestRule.activity)
+        webViewUtils.waitForElement("#catroid-catblocks-container") {
+            webViewUtils.waitForElement("#catroid-catblocks-container > div > svg > g")
+        }
+
+        UiTestCatroidApplication.projectManager.currentSprite.getScript(0).posX = 50f
+        UiTestCatroidApplication.projectManager.currentSprite.getScript(0).posY = 50f
+        Assert.assertEquals(
+            UiTestCatroidApplication.projectManager.currentSprite.getScript(0).posX,
+            50.0f
+        )
+        Assert.assertEquals(
+            UiTestCatroidApplication.projectManager.currentSprite.getScript(0).posY,
+            50.0f
+        )
+    }
+
+    private fun createProject() {
+        val projectName = javaClass.simpleName
+        val project = UiTestUtils.createDefaultTestProject(projectName)
+        val sprite = Sprite("testSprite")
+        project.defaultScene.addSprite(sprite)
+        val startScript: Script = StartScript()
+        val ifBrick = IfLogicBeginBrick()
+        ifBrick.addBrickToIfBranch(SetXBrick())
+        ifBrick.addBrickToElseBranch(ChangeXByNBrick())
+        startScript.addBrick(ifBrick)
+        startScript.setParents()
+        sprite.addScript(startScript)
+        UiTestCatroidApplication.projectManager.currentProject = project
+        UiTestCatroidApplication.projectManager.currentSprite = sprite
```

Comment:
No need to create a new Sprite and Script, UiTestUtils.createDefaultTestProject(projectName) did this already for you.
To acess them just use project.defaultScene.spriteList[1]

Are these extra bricks really necessary, because you test only the position of the startScript?
If not place remove them.